hedera-node/docs/design/services/smart-contract-service/contract-accounts-nonces.md

```diff
@@ -0,0 +1,64 @@
+| EOA transaction (`EthereumTransaction`)                                                 | EOA nonce is incremented                                                                                   | EOA nonce is incremented, updates are not externalized to Mirror Node through transaction records, but can be picked up by reading the value from the `EthereumTransaction` body | EOA nonce is incremented, updates are not externalized to Mirror Node through transaction records, but can be picked up by reading the value from the `EthereumTransaction` body |
+| EOA transaction (`ContractCall` or `ContractCreate`)                                    | -                                                                                                          | -                                                                                                                                                                                | -                                                                                                                                                                                |
+| Contract transaction resulting in `CREATE/CREATE2` (`EthereumTransaction`)              | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account, but updates are not externalized to Mirror Node                      | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account, but updates are externalized to Mirror Node                          |
+| Contract transaction resulting in `CREATE/CREATE2` (`ContractCall` or `ContractCreate`) | -                                                                                                          | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account, but updates are not externalized to Mirror Node                      | initial contract nonce value is 1; nonce is incremented for each contract creation initiated by an account, but updates are externalized to Mirror Node                          |
+
+
+
+### Contract Nonce Externalization
+
+- We can keep a `ContractId -> nonce` map inside `TxnAwareRecordsHistorian`, it would be updated on each call to `AbstractRecordingCreateOperation` (possibly using its `createOperationExternalizer` field) even if the contract creation operation does not succeed
```

Comment:
Q: to be clear the map will only contain contracts that create in the given transaction correct? It's not a case of having to consider or filter through all contracts in state to see the new ones

Reply:
> to be clear the map will only contain contracts that create in the given transaction correct?

Yes, that it correct.

> It's not a case of having to consider or filter through all contracts in state to see the new ones

Not sure if I understand this correctly, can you elaborate?

Reply:
If the first statement is true then no worries.
I just wanted to make sure that in determining the new contract nonces we weren't looping through all contracts and were only considering contracts touched in this transaction